modules/javascript/partytown-web-worker/load.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Module Name: Partytown Web Worker
+ * Description: Run JavaScript in a separate Web Worker with the help of Partytown library.
+ * Experimental: Yes
+ *
+ * @since   n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * PartyTown Configuration
+ *
+ * @since n.e.x.t
+ * @see https://partytown.builder.io/configuration
+ * @return array
+ */
+function perflab_partytown_web_worker_configuration() {
+	$plugin_dir           = trailingslashit( plugin_dir_path( __FILE__ ) );
+	$content_dir_basename = basename( WP_CONTENT_DIR );
+
+	$config = array(
+		'lib'     => '/' . substr( $plugin_dir, strpos( $plugin_dir, $content_dir_basename ) ) . 'assets/js/partytown/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for PartyTown.
+	 *
+	 * @since n.e.x.t
+	 * @see <https://partytown.builder.io/configuration>.
+	 * @param array $config Configuration for PartyTown.
+	 * @return array
+	 */
+	return apply_filters( 'perflab_partytown_configuration', $config );
+}
+
+/**
+ * Initialize PartyTown
+ *
+ * @since n.e.x.t
+ * @return void
+ */
+function perflab_partytown_web_worker_init() {
+	$partytown_js = __DIR__ . '/assets/js/partytown/partytown.js';
+
+	if ( file_exists( $partytown_js ) ) {
```

Comment:
If the file doesn't exist, shouldn't this do `wp_die()` instead of add an empty string? This file not existing indicates a development issue which wouldn't occur in production.

Reply:
That makes sense. and maybe with a notice?

Reply:
A notice in `wp_die()`? Ideally we wouldn't include the `file_exists()` check in the build of the plugin, since we wouldn't need to keep checking for it all the time as it would be guaranteed to be there. The condition could be stripped out for distribution. This is something that could be done in [`build-plugins`](https://github.com/WordPress/performance/blob/0e7f28516e452e107fbaf66a3a548c274f230878/bin/plugin/commands/build-plugins.js#L67-L94), although that is for standalone _plugins_ and not modules. In any case, I don't think that this should necessarily hold up this PR. There should just be some error communicated that the user hasn't done a build.

Reply:
Agree since FS checks are costly. To address this:
- Add logic to not load Partytown module at all if vendor files does not exists and add admin notice - https://github.com/WordPress/performance/pull/556/commits/8c76f52159f0b386dc4e026e9dc2080f3eae99b9
- Remove file exists check for Partytown library - https://github.com/WordPress/performance/pull/556/commits/16a466c7f5181fda8f3ff3af2139ffe821fc8b6b